### PR TITLE
initial setup for SHACL based validation

### PR DESCRIPTION
**Pull request by chenejac**
**[VIVO GitHub issue](https://github.com/vivo-project/VIVO/issues)**: [3665](https://github.com/vivo-project/VIVO/issues/3665)

# What does this pull request do?
This PR is introducing an engine for validation of dynamic actions and resourceAPIs over defined SHACL rules. 

# What's new?
SHACL rules for validation has been defined (but will be refined in some future PRs). Validation of dynamic actions and resourceAPIs might be done:

- Before loading into the graph base. This is not turned-on yet in regular Vitro application, but is tested by a [unit test](https://github.com/vivo-project/Vitro/blob/4289d41663bf54358f5c6ac58567b59242a590d2/api/src/test/java/edu/cornell/mannlib/vitro/webapp/dynapi/ActionPoolTest.java#L245). See also [this](https://github.com/vivo-project/Vitro/blob/4289d41663bf54358f5c6ac58567b59242a590d2/api/src/test/java/edu/cornell/mannlib/vitro/webapp/dynapi/ServletContextTest.java#L90)
-  As part of [bean loading process](https://github.com/vivo-project/Vitro/blob/4289d41663bf54358f5c6ac58567b59242a590d2/api/src/main/java/edu/cornell/mannlib/vitro/webapp/utils/configuration/ConfigurationBeanLoader.java#L180). This is also tested by a [unit test](https://github.com/vivo-project/Vitro/blob/4289d41663bf54358f5c6ac58567b59242a590d2/api/src/test/java/edu/cornell/mannlib/vitro/webapp/dynapi/ActionPoolTest.java#L265). 

# How should this be tested?
New functionality is covered by three unit tests in the ActionPoolTest. Also, RESTEndpointIntegrationTest is enriched to try to load a not valid Action, which is not available through REST endpoint (because BeanLoader is not loading that in the Pool as an invalid action and resourceAPI), although it is loaded in the [graph](https://github.com/vivo-project/Vitro/blob/4289d41663bf54358f5c6ac58567b59242a590d2/api/src/test/resources/dynapi/mock/rest/response/body/get/test_collection_resource.json#L1). 

# Additional Notes:
Any additional information that you think would be helpful when reviewing this PR.

* Does this change add any new dependencies? Yes, Topbraid SHACL library is used for validation - https://mvnrepository.com/artifact/org.topbraid/shacl 

# Interested parties
Tag (@ mention) interested parties or, if unsure, @VIVO-project/vivo-committers

### CODE REVIEW

File: api/src/main/java/edu/cornell/mannlib/vitro/webapp/utils/configuration/ConfigurationBeanLoader.java [hunk 1]
```diff
@@ -206,7 +223,7 @@ public <T> Set<T> loadEach(Class<T> resultClass){
 			try {
 				instances.add(loadInstance(uri, resultClass));
 			} catch (ConfigurationBeanLoaderException e) {
-				e.printStackTrace();
```
Comment (litvinovg) [inline api/src/main/java/edu/cornell/mannlib/vitro/webapp/utils/configuration/ConfigurationBeanLoader.java:209]: Why remove printStackTrace ?

**Comment by litvinovg** [comment]
I suggest to always use curly braces and I would create a package for model validators

File: api/src/main/java/edu/cornell/mannlib/vitro/webapp/utils/configuration/ConfigurationBeanLoader.java (hunk 1 above)
Comment (chenejac) [reply api/src/main/java/edu/cornell/mannlib/vitro/webapp/utils/configuration/ConfigurationBeanLoader.java:209]: ok, I will fix this.

**Comment by chenejac** [comment]
superseded by https://github.com/vivo-project/Vitro/pull/364

**Pull request closed without merging**